engine/execution/provider/engine.go

```diff
@@ -371,7 +358,7 @@ func (e *Engine) ensureAuthorized(chunkID flow.Identifier, originID flow.Identif
 	}
 
 	if origin.Weight == 0 {
```

Comment:
## ⚠️ 

this will undermine liveness:
* imagine we just finalized the first block of the epoch 
* but sealing is still in the last epoch
* multiple of the assigned verification nodes are no longer part of the new epoch, but still try approve unsealed blocks from the last epoch
* Execution nodes will not answer their query => block does not get approved => game over!

_Suggestion_
* we don't need to answer queries from _ejected nodes_ 
* but we must answer queries from weight-zero nodes 

```suggestion
	if origin.Ejected {
		return nil, engine.NewInvalidInputErrorf("node %s was ejected and therefore does not receive data", originID)
	}
```


Additional context and notes:
* With your change to using a protocol state from the finalized block, the weight might be different than the one that was used for the verifier assignment. 
* Note: your change makes it really obvious that we also had an uncovered edge case also in the previous code: previously we used the list of nodes that were authorized as of the executed block.  However, the chunk assigner uses the authorized nodes as of the block that _incorporates_ the result (-> [code](https://github.com/onflow/flow-go/blob/dfd6ddb799d29811e493c82fae41cd68256278bd/engine/consensus/approvals/verifying_assignment_collector.go#L145))
      
   For example the last block of an epoch illustrates this well. The nodes that are authorized as of this block are the nodes in the old epoch. While the result is incorporates in a block of the new epoch.